File: dataset/human/python/file_2343.py
import math, sys
sys.setrecursionlimit(10**7)
d,g = map(int, input().split())
data = []
ans = 10000000
for i in range(1,d+1):
    p, c = map(int, input().split())
    data.append([i*100 , p , c])
def dfs(li):
    if len(li) == d:
        global ans
        total = 0
        candidate = 0
        for i in range(d):
            if li[i]:
                total += data[i][0] * data[i][1] + data[i][2]
                candidate += data[i][1]
        if total >= g:
            ans = min(ans, candidate)
        else:
            max_point = max([data[i] for i in range(d) if not li[i]])
            if max_point[0] * (max_point[1] - 1) + total >= g:
                candidate += math.ceil((g - total) / max_point[0])
                ans = min(ans, candidate)
    else:
        dfs(li + [False])
        dfs(li + [True])
dfs([])
print(ans)

File: dataset/human/c/file_2343.c
#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include<math.h>

#define sl(x) scanf("%ld", &x)
#define ss(x) scanf("%s", x)
#define pl(x) printf("%ld\n", x)
#define pd(x) printf("%lf\n", x)
#define ps(x) printf("%s\n", x)

#define INF 1000000000000000000
#define MOD 1000000007

#define MIN(a,b) (((a)<(b))?(a):(b))
#define MAX(a,b) (((a)>(b))?(a):(b))

int main() {
  long N, K, S;
  long i, j;
  long ans = 0;
  
  sl(N); sl(K); sl(S);
  if(N == K) {
    for(i=0;i<N;i++) {
      if(i < N-1) printf("%ld ", S);
      else printf("%ld\n", S);
    }
  } else if(S == 1) {
    for(i=0;i<K;i++) {
      printf("1 ");
    }
    for(i=K;i<N;i++) {
      if(i < N-1) printf("1000000000 ");
      else printf("1000000000\n");
    }
  } else {
    for(i=0;i<K+1;i++) {
      if(i%2 == 0) printf("%ld ", S-1);
      else printf("1 ");
    }
    if(S == 1000000000) {
      for(i=K+1;i<N;i++) {
        if(i < N-1) printf("500000001 ");
        else printf("500000001\n");
      }
    } else {
      for(i=K+1;i<N;i++) {
        if(i < N-1) printf("1000000000 ");
        else printf("1000000000\n");
      }
    }
  }
  
  return 0;
}
